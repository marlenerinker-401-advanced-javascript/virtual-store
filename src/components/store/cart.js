// Our global state object just for votes
const initialState = [];

    


// Reducers, a function that takes an action and produces a new version of state, from a payload and a type.
export default (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {
  case 'ADD':
    
    return [...state, payload];

  case 'REMOVE':
    // eslint-disable-next-line no-case-declarations
    let newCart = state;
    for(let i = 0; i<newCart.length; i++) {
      if(newCart[i].name === payload.name){
        newCart.splice(i, 1);
      }
      return [...newCart];
    }

    return newCart;
    
    


  default:
    return state;
  }

  
  

};


// actions are the functions that components can run themselves
export const addItem = (product) => {
  return {
    type: 'ADD',
    payload: product,
  };
};

export const removeItem = (product) => {
  return {
    type: 'REMOVE',
    payload: product,
  };
};
