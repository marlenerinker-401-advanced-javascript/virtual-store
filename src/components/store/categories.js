/* eslint-disable no-case-declarations */
// Our global state object just for votes
const initialState = {
  categories: [
    { normalizedName: 'categoryone', displayName: 'Category One', description: 'category one' },
    { normalizedName: 'categorytwo', displayName: 'Category Two ', description: 'category two' },
    { normalizedName: 'categorythree', displayName: 'Category Three', description: 'category three' },
    { normalizedName: 'categoryfour', displayName: 'Category Four', description: 'category four' },
  ],
  activeCategory: {displayName: 'No category selected'},
};

// Reducers, a function that takes an action and produces a new version of state, from a payload and a type.
export default (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {
  case 'CHANGE':
  
    return {...state, activeCategory: payload};

  default:
    return state;
  }
};


// actions are the functions that components can run themselves
export const changeCategory = (category) => {
  return {
    type: 'CHANGE',
    payload: category,
  };
};

