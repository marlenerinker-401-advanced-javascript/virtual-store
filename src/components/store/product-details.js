/* eslint-disable no-case-declarations */
import axios from 'axios';





// Our global state object just for votes
const initialState = [];
   


// Reducers, a function that takes an action and produces a new version of state, from a payload and a type.
export default (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {

  case 'GET_PRODUCT':
    console.log('get product payload: ', payload);
    return payload;

  default:
    return state;
  } 
  

};


// actions are the functions that components can run themselves

export const getOneProduct = (product) => async (dispatch) => {
  const response = await axios.get(`http://localhost:3000/products/${product.id}`);
  dispatch({
    type: 'GET_PRODUCT',
    payload: response.data,
  });
};

