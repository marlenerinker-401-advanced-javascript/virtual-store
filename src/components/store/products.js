/* eslint-disable no-case-declarations */



// Our global state object just for votes
const initialState = {
  products: [
    { category: 'categoryone', name: 'Product One', description: 'fun product', price: '$5.00', inventory: 0},
    { category: 'categoryone', name: 'Product Two', description: 'fun product', price: '$5.00', inventory: 0},
    { category: 'categorytwo', name: 'Product Three', description: 'fun product', price: '$5.00', inventory: 0},
    { category: 'categorytwo', name: 'Product Four', description: 'fun product', price: '$5.00', inventory: 0},
    { category: 'categorythree', name: 'Product Five', description: 'fun product', price: '$5.00', inventory: 0},
    { category: 'categorythree', name: 'Product Six', description: 'fun product', price: '$5.00', inventory: 0},
    { category: 'categoryfour', name: 'Product Seven', description: 'fun product', price: '$5.00', inventory: 0},
    { category: 'categoryfour', name: 'Product Eight', description: 'fun product', price: '$5.00', inventory: 0},
  ],
};

// Reducers, a function that takes an action and produces a new version of state, from a payload and a type.
export default (state = initialState, action) => {
  const { type, payload } = action;

  switch (type) {
  case 'CHANGE':
    // for (let i = 0; i < state.products.length; i++) {
    //   if (state.products[i].category === payload) {
    //     return {...state, isActive: true};
    //   }
    // }
    console.log('product change is running', payload);
    
    return state;

  default:
    return state;
  }

  
  

};


// actions are the functions that components can run themselves
// export const changeCategory = (category) => {
//   return {
//     type: 'CHANGE',
//     payload: category,
//   };
// };

