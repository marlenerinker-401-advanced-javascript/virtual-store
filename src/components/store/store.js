import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import categories from './categories.js';
import products from './products.js';
import cart from './cart.js';

let reducers = combineReducers({ categories, products, cart });

let store = () => {
  return createStore(reducers, composeWithDevTools(applyMiddleware(thunk)));
};

export default store();