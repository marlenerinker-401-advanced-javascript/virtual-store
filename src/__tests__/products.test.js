import React from 'react';

import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'

import Products from '../components/layout/products.js';
import { Provider } from 'react-redux';
import store from '../components/store/store.js';

describe('Product tests', () => {

  it('renders correctly', () => {
    const DOM = renderer.create(<Provider store={store}><Products/></Provider>).toJSON();
    expect(DOM).toMatchSnapshot();
  });

});