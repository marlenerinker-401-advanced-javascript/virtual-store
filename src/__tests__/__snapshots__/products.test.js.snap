// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product tests renders correctly 1`] = `
<div
  className="makeStyles-productsRoot-1"
>
  <div
    className="MuiPaper-root MuiPaper-outlined MuiPaper-rounded"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4"
      id="product-title"
    >
      Products
    </h4>
    <ul
      className="MuiList-root MuiList-padding"
    />
  </div>
</div>
`;
